src/runtime/virtcontainers/network_linux.go

```diff
@@ -597,7 +597,12 @@ func xConnectVMNetwork(ctx context.Context, endpoint Endpoint, h Hypervisor) err
 	queues := 0
 	caps := h.Capabilities(ctx)
 	if caps.IsMultiQueueSupported() {
-		queues = int(h.HypervisorConfig().NumVCPUs())
+		queues = int(h.HypervisorConfig().NumTapQueues)
+		if (queues <= 0) {
+			queues = int(h.HypervisorConfig().NumVCPUs())
+		} else if queues > int(h.HypervisorConfig().DefaultMaxVCPUs) {
+			queues = int(h.HypervisorConfig().DefaultMaxVCPUs)
+		}
```

Comment:
I am not sure if we should provide a detection mechanism to guarantee that the network queues entered by the user are correctly linked to the number of CPUs in the VM, to ensure that the number of network queues set by the user is valid, and that the final result of the multi-queue setting meets expectations. If we do so, maybe, we should do it during the `load_config` process, or in some other way.

Reply:
About the `<= 0` is to maintain behaviour, meaning that with the default value (0) it'll behave exactly as it did before, meaning that the number of queues will be set to the number of vCPUs.

About `> DefaultMaxVCPUs`, I was also not sure about this, but I saw that kata does ensure for example that DefaultVCPUs doesn't exceed the number of physical core of the system. I found similar reasoning with this limitation, meaning that setting a number of queues greater than the number of cores available to the VM it's a waste of resources without benefit. Do you think I should remove this?

Anyway, I'll refactor this at config loading time.